config/esp32/components/chip/Kconfig

```diff
@@ -880,14 +873,35 @@ menu "CHIP Device Layer"
         config MAX_PERMIT_LIST_SIZE
             int "Set permit list size for Insights traces"
             range 5 30
-            depends on ESP_INSIGHTS_ENABLED || ENABLE_ESP_DIAGNOSTICS_TRACE
+            depends on ESP_INSIGHTS_ENABLED
             default 20
             help
                 Set the maximum number of group entries that can be included in the permit list for reporting
                 traces to Insights or diagnostics. This ensures proper management of trace reporting capacity.
     endmenu
 
+    menu "Diagnostics Options"
+        config ENABLE_ESP_DIAGNOSTICS_TRACE
+            bool "Enable ESP Platform Diagnostics for Matter"
+            default n
+            help
+                Enables the ESP Diagnostics platform to collect, store, and retrieve diagnostic data for the Matter protocol.
+                This feature helps monitor system health and performance by providing insights through diagnostics logs.
 
+        config ENABLE_TRACES
+            bool "Enable Traces"
+            default y
+            depends on ENABLE_ESP_DIAGNOSTICS_TRACE
+            help
+                Enable Traces for the ESP Platform Diagnostics for Matter.
+
+        config ENABLE_METRICS
+            bool "Enable Metrics"
+            default y
+            depends on ENABLE_ESP_DIAGNOSTICS_TRACE
```

Comment:
Why metrics option depends on the `ENABLE_ESP_DIAGNOSTICS_TRACE`? Should the above one be only `ENABLE_ESP_DIAGNOSTICS`?